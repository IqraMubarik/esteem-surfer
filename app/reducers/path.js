// @flow
import {LOCATION_CHANGE} from 'react-router-redux'


type actionType = {
    type: string,
    +payload?: {}
};

const defaultState = {
    prev: null,
    cur: null,
};


export default function path(state = defaultState, action: actionType) {
    switch (action.type) {
        case LOCATION_CHANGE:
            return {
                prev: state.cur,
                cur: action.payload.pathname,
            };
        default:
            return state
    }
}