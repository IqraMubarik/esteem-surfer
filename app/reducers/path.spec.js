/* eslint-disable */
import path from './path';
import {LOCATION_CHANGE} from 'react-router-redux'
import deepFreeze from 'deep-freeze';


describe('path reducer', () => {
    it('should handle initial state', () => {
        expect(path(undefined, {})).toMatchSnapshot();
    });

    it('should handle LOCATION_CHANGE', () => {
        const stateBefore = {
            prev: null,
            cur: null
        };

        deepFreeze(stateBefore);

        const action = {
            type: LOCATION_CHANGE,
            payload: {
                hash: "",
                pathname: "/trending",
                search: "",
                state: undefined
            }
        };

        const stateAfter = {
            prev: null,
            cur: "/trending"
        };

        const res = path(stateBefore, action);
        expect(res).toEqual(stateAfter);
    });

    it('should handle LOCATION_CHANGE', () => {
        const stateBefore = {
            prev: null,
            cur: "/trending"
        };

        deepFreeze(stateBefore);

        const action = {
            type: LOCATION_CHANGE,
            payload: {
                hash: "",
                pathname: "/trending/art",
                search: "",
                state: undefined
            }
        };

        const stateAfter = {
            prev: "/trending",
            cur: "/trending/art"
        };

        const res = path(stateBefore, action);
        expect(res).toEqual(stateAfter);
    });
});

