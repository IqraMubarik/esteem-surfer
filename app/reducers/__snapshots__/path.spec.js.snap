// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`path reducer should handle initial state 1`] = `
Object {
  "cur": null,
  "prev": null,
}
`;
